mmv1/templates/terraform/custom_expand/sd_full_url.tmpl

```diff
@@ -4,7 +4,8 @@ func expand{{$.GetPrefix}}{{$.TitlelizeProperty}}(v interface{}, d tpgresource.T
   } else if strings.HasPrefix(v.(string), "https://") {
     return v, nil
   }
-  url, err := tpgresource.ReplaceVars(d, config, "{{"{{"}}ServiceDirectoryBasePath{{"}}"}}" + v.(string))
+  // v1 is the only version in use 
+  url, err := tpgresource.ReplaceVars(d, config, "https://servicedirectory.googleapis.com/v1/" + v.(string))
```

Comment:
Could we substitute `v1` for `beta` specifically or confirm this service isn't in alternate universes rather than providing a fixed URL here? Fixed URLs are mostly unsafe to use in checks nowadays, unfortunately!

Reply:
Good point. It's necessary to do a substitution after the fact so we capture BasePath changes that have been applied before this code runs.

Reply:
I believe substituting v1 for beta should be working in this case, I assume you mean something similarly to https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/tpgresource/self_link_helpers.go#L95. Yeah, that's safer than using fixed URLs

Reply:
This approach works in GCP! I meant a substring replacement of `v1` for `beta` rather than a fixed URL prefix like we're currently doing*

Reply:
Sure, I'll revert this change and create a new one

Reply:
Thanks! Sorry 😅 

@c2thorn we may want to add a lint rule against fixed GCP URLs appearing in the provider outside of the specific spots they're safe to appear

Reply:
Issue: https://github.com/hashicorp/terraform-provider-google/issues/21266